src/include/settings/settings_manager.h

```diff
@@ -0,0 +1,139 @@
+#pragma once
+
+#include <gflags/gflags.h>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <utility>
+#include "catalog/settings_handle.h"
+#include "common/action_context.h"
+#include "common/exception.h"
+#include "common/shared_latch.h"
+#include "loggers/settings_logger.h"
+#include "settings/settings_param.h"
+#include "type/transient_value.h"
+#include "type/transient_value_peeker.h"
+
+namespace terrier {
+class DBMain;
+}
+
+namespace terrier::settings {
+using setter_callback_fn = void (*)(const std::shared_ptr<common::ActionContext> &action_context);
+
+/**
+ * A wrapper for pg_settings table, does not store values in it.
+ * Stores and triggers callbacks when a tunable parameter is changed.
+ * Static class.
+ */
+
+class SettingsManager {
+ public:
+  SettingsManager() = delete;
+  SettingsManager(const SettingsManager &) = delete;
+
+  /**
+   * The constructor of settings manager
+   * @param db a pointer to the DBMain
+   * @param catalog a shared pointer to the system catalog
+   * @param txn_manager a pointer to the transaction manager
+   */
+  SettingsManager(DBMain *db, catalog::Catalog *catalog, transaction::TransactionManager *txn_manager);
+
+  /**
+   * Get the value of an integer setting
+   * @param param setting name
+   * @return current setting value
+   */
+  int32_t GetInt(Param param);
+
+  /**
+   * Get the value of a double setting
+   * @param param setting name
+   * @return current setting value
+   */
+  double GetDouble(Param param);
+
+  /**
+   * Get the value of a boolean setting
+   * @param param setting name
+   * @return current setting value
+   */
+  bool GetBool(Param param);
+
+  /**
+   * Get the value of a string setting
+   * @param param setting name
+   * @return current setting value
+   */
+  std::string_view GetString(Param param);
```

Comment:
I'm worried about potential lifetime issues here. `string_view`s themselves do not own underlying memory. But here you are giving out the reference, which is a dangerous thing to do. Our transient value implementation returns `string_view`s mainly because we relies on programmers to manage lifetime issues. I don't know whether the `string_view` here is safe.

Reply:
Thank you. Solved.